source/ada/lsp-ada_driver.adb

```diff
@@ -79,6 +84,7 @@ begin
 
       begin
          if Do_Exit then
+            Trying_To_Finalize := True;
             Server.Finalize;
             return;
```

Comment:
Quite dangerous for me :) I suggest to move this code outside of the loop.

Reply:
done